avaframe/in3Utils/cfgHandling.py

```diff
@@ -0,0 +1,364 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+'''
+    Utilities for working with cfg info
+'''
+
+import logging
+import numpy as np
+
+# Local imports
+from avaframe.in3Utils import cfgUtils
+import avaframe.in3Utils.fileHandlerUtils as fU
+
+log = logging.getLogger(__name__)
+
+
+def insertIntoSimName(name, keys, values, index):
+    """ Add keys and values to name, in between parts of name split by index
+
+        Parameters
+        -----------
+        name: str
+            name to extend
+        keys: list
+            list with keys
+        values: list
+            list with values
+        index: str
+            used to split name
+
+        Returns
+        --------
+        newName: string
+            containing newName, with keys and values inserted after index
+    """
+
+    # Split according to index
+    splitName = name.split(index + '_')
+    newPart = '_'
+
+    # Loop through keys
+    for key, value in zip(keys, values):
+        newPart = newPart + str(key) + '_' + str(value) + '_'
+
+    # Put newname back together
+    try:
+        newName = splitName[0] + str(index) + newPart + splitName[1]
+    except IndexError:
+        log.info(splitName)
+        log.error('Some part is missing. SOMENAME_simHash_XXX is expected')
+        raise
```

Comment:
this will raise the IndexError message not your custom one. 
You could do IndexError(your message to have both)